### PR TITLE
Add parameterization for benchmark flag

### PR DESCRIPTION
**Pull request by erman-gurses**

### CODE REVIEW

File: tests/kernel/wave/wave_gemm_test.py [hunk 1]
```diff
@@ -137,5 +145,5 @@ def repeat(acc: tkl.Register[M, N, tkl.f32]) -> tkl.Register[M, N, tkl.f32]:
                 f.write(mb.module_op.get_asm())
 
         iree_ref = torch.zeros(shape[0], shape[1], dtype=torch.float32)
-        generate_iree_ref("mmt", [a, b], [iree_ref], config)
+        generate_iree_ref("mmt", [a, b], [iree_ref], config, run_bench=True)
```
Comment (Hardcode84) [inline tests/kernel/wave/wave_gemm_test.py:148]: `run_bench=run_bench`

File: tests/kernel/wave/wave_gemm_test.py [hunk 2]
```diff
@@ -137,5 +145,5 @@ def repeat(acc: tkl.Register[M, N, tkl.f32]) -> tkl.Register[M, N, tkl.f32]:
                 f.write(mb.module_op.get_asm())
 
         iree_ref = torch.zeros(shape[0], shape[1], dtype=torch.float32)
```
Comment (Hardcode84) [inline tests/kernel/wave/wave_gemm_test.py:147]: you also need to add `benchmark_batch_size`, `benchmark_repetitions` and `benchmark_results_file` like in igemm test.

Comment (erman-gurses) [reply tests/kernel/wave/wave_gemm_test.py:147]: I can add that. Are those numbers good?
        config["benchmark_batch_size"] = 10
        config["benchmark_repetitions"] = 3

Comment (Hardcode84) [reply tests/kernel/wave/wave_gemm_test.py:147]: Yes, lets go with these value for now.

File: tests/kernel/wave/wave_gemm_test.py [hunk 3]
```diff
@@ -136,6 +154,11 @@ def repeat(acc: tkl.Register[M, N, tkl.f32]) -> tkl.Register[M, N, tkl.f32]:
...
-        generate_iree_ref("mmt", [a, b], [iree_ref], config)
-        assert_close(c, iree_ref)
+        if run_bench:
+            if dump_perf is not None:
+                config["benchmark_results_file"] = os.path.join(
+                    dump_perf, "iree_" + perf_filename
+                )
+            iree_ref = torch.zeros(shape[0], shape[1], dtype=torch.float32)
+            generate_iree_ref("mmt", [a, b], [iree_ref], config, run_bench=run_bench)
+            assert_close(c, iree_ref)
```
Comment (Hardcode84) [inline tests/kernel/wave/wave_gemm_test.py:164]: This check is needed for correctness, it should present bot in `run_bench=True` and `run_bench=False` paths

**Review by Hardcode84 — APPROVED** [review]

**Pull request merged**